addons/mass_mailing/models/mailing.py

```diff
@@ -58,6 +58,16 @@ def _get_default_mail_server_id(self):
     email_from = fields.Char(string='Send From', required=True,
         default=lambda self: self.env.user.email_formatted)
     sent_date = fields.Datetime(string='Sent Date', copy=False)
+
+    def _default_schedule_type(self):
+        default_schedule_date = self.env.context.get('default_schedule_date')
+        if default_schedule_date:
+            return "send_on"
+        return "send_asap"
+
+    schedule_type = fields.Selection(
+        [('send_asap', 'Send ASAP'), ('send_on', 'Send on')], string='Schedule', default=_default_schedule_type
```

Comment:
Not sure how "ASAP" will translate in other languages.
What about "Send now" ?

Not a big fan of the naming in general.
Here is how we do it in social for the same concept, I think it's a bit more appropriate overall:
```py
post_method = fields.Selection([
        ('now', 'Send now'),
        ('scheduled', 'Schedule later')], string="When", default='now', required=True,
        help="Publish your post immediately or schedule it at a later time.")
```